crates/wit-parser/src/ast.rs

```diff
@@ -226,6 +229,42 @@ impl<'a> ExternKind<'a> {
     }
 }
 
+pub struct Include<'a> {
+    from: UsePath<'a>,
+    names: Vec<UseName<'a>>,
+}
+
+impl<'a> Include<'a> {
+    fn parse(tokens: &mut Tokenizer<'a>) -> Result<Self> {
+        tokens.expect(Token::Include)?;
+        let from = UsePath::parse(tokens)?;
+
+        let mut names = Vec::new();
+
+        match tokens.clone().next()? {
+            Some((_span, Token::With)) => {
+                tokens.expect(Token::With)?;
+                tokens.expect(Token::LeftBrace)?;
+                while !tokens.eat(Token::RightBrace)? {
+                    let mut name = UseName {
+                        name: parse_id(tokens)?,
+                        as_: None,
+                    };
+                    if tokens.eat(Token::As)? {
+                        name.as_ = Some(parse_id(tokens)?);
+                    }
```

Comment:
I realize this is how `use` is parsed but I think the `as` keyword should be required for `with`?